api/ScreenOrientation.json

```diff
@@ -103,25 +103,41 @@
           "spec_url": "https://w3c.github.io/screen-orientation/#dom-screenorientation-lock",
           "support": {
             "chrome": {
-              "version_added": "38"
+              "version_added": "38",
+              "partial_implementation": true,
+              "notes": "Always throws <code>NotSupportedError</code>."
             },
             "chrome_android": {
               "version_added": "38"
             },
             "edge": {
-              "version_added": "79"
+              "version_added": "79",
+              "partial_implementation": true,
+              "notes": "Always throws <code>NotSupportedError</code>."
             },
             "firefox": {
-              "version_added": "43"
-            },
-            "firefox_android": {
-              "version_added": "43"
-            },
+              "version_added": "43",
+              "partial_implementation": true,
+              "notes": "Always throws <code>NotSupportedError</code>."
+            },
+            "firefox_android": [
+              {
+                "version_added": "79",
+                "partial_implementation": true,
+                "notes": "The API exists but does not lock the screen orientation."
```

Comment:
Is this a different behavior than throwing `NotSupportedError`?

Reply:
`NotSupportedError` is thrown on Firefox Android 97+.  Android 79-96 will throw `NS_ERROR_UNEXPETED`

Reply:
> Is this a different behavior than throwing NotSupportedError?

Sorry for the late response - just back from holiday.

@ddbeck What I was saying here is that in my testing of FF **android** prior to FF96 I could call the API but the orientation would simply not lock - no error or anything. This differs from the desktop version which throws `NotSupportedError` to indicate that locking is not supported.

@makotokato Are you saying that my testing was incorrect - there is a thrown error in **FF Android 79-96** of `NS_ERROR_UNEXPECTED` if you tried to use the API on android?

@ddbeck If so, in that case wouldn't we just assume that this is the default case for anything not supported (i.e. "equivalent" to `NotSupportedError` and simply remove this support statement. Then it would look like this was supported from 43 and removed in 79, which is in fact the way users will see it.

Otherwise I can put this error in.

> `NotSupportedError` is thrown on Firefox Android 97+.

Presumably only if the preference is not enabled.

Reply:
> Are you saying that my testing was incorrect - there is a thrown error in FF Android 79-96 of NS_ERROR_UNEXPECTED if you tried to use the API on android?

I tested this by https://makotokato.github.io/demo/orientation.html. Firefox 96 throws `NS_ERROR_UNEXPECTED`. Firefox (beta) 97+ throws `NotSupportedError` if pref isn't set.

Reply:
Thanks @makotokato. So @ddbeck Above, question is whether we try to capture that we get `NS_ERROR_UNEXPECTED` or we just pretend that support was removed from FF79 (ie mark that as version things were removed).

Reply:
Unfortunately, convention requires that we represent this API as partially supported. Setting a `version_removed` can confuse things a bit, since a simple feature detection will show that the feature exists (i.e., `"lock" in ScreenOrientation.prototype` returns `true`) though it doesn't actually _do_ anything.

Reply:
@ddbeck I don't understand the action you are asking me to take here ^^^ with `lock`. 

This is how I see it - but I could be wrong - I've lost track over time and I'm not sure I saw the preference first time around. @makotokato would appreciate your confirmation.

Firefox **desktop** looks as below
- The feature has existed since 43 but always throws `NotSupportedError` on desktop. 
- There is a pref (`dom.screenorientation.allow-lock`) which was introduced at v97 (I think), but this is _irrelevant_ because whether true or false you get `NotSupportedError` on desktop

```json
            "firefox": {
              "version_added": "43",
              "partial_implementation": true,
              "notes": "Always throws <code>NotSupportedError</code>."
            },
```

Firefox android
-  has existed since FF43 and worked in the early versions..
- In v79 (first version on new UI framework) it was broken. From FF79 to FF96 it returns NS_ERROR_UNEXPECTED 
- FF97 the API was fixed if `dom.screenorientation.allow-lock` is true. If not set, it returns `NotSupportedError`.
- But as per MDN standard, the pref is considered not to exist on Android.

This is what the definition looks like right now. 
```json
            "firefox_android": [
              {
                "version_added": "79",
                "partial_implementation": true,
                "notes": "The API exists but does not lock the screen orientation."
              },
              {
                "version_added": "43",
                "version_removed": "79"
              }
            ],
```
I think this might be wrong. It is supposed to indicate that this API works from 43 to 79. Then from 79 it should indicate that it exists but doesn't work correctly. Then in FF97 it should appear removed, because the preference cannot be set. Is this what you meant?

```json
            "firefox_android": [
              {
                "version_added": "79",
                "version_removed": "97",
                "partial_implementation": true,
                "notes": "The API exists but returns <code>NS_ERROR_UNEXPECTED</code> ."
              },
              {
                "version_added": "43",
                "version_removed": "79"
              }
            ],
```

Reply:
Yes, this last code block would be correct. Sorry for misunderstanding before.

Reply:
No worries. I don't think I'd got to the point where this could be put succinctly before :-)